### PR TITLE
[KYUUBI #476][FOLLOWUP] Strip prefixes when ignoring and restricting configs

### PR DESCRIPTION
**Pull request by yaooqinn**
yaooqinn [Closes #510](https://github.com/yaooqinn/kyuubi/pull/510)    [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

1. change the configs to seq[string] for simplification 
2. bugfix for mismatched configs when  ignoring and restricting because they are maybe prefixed with something like `set:hivevar:thekey=thevalue`, we should normalize it to `thekey=thevalue`

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @pan3793 @turboFei @jiwq PTAL

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 1]
```diff
@@ -97,6 +97,43 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
...
+        newKey.substring(METACONF_PREFIX.length)
+      } else {
+        newKey
+      }
+    } else {
+      key
+    }
+
+    if (_confRestrictList.contains(normalizedKey)) {
+      throw KyuubiSQLException(s"$normalizedKey is a restrict key according to the sever side" +
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:123]: typo: sever

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 2]
```diff
@@ -53,26 +53,7 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
     val handle = sessionImpl.handle
     try {
       val sparkSession = spark.newSession()
-      conf.foreach {
-        case (k, v) if k.startsWith(SET_PREFIX) =>
-          val newKey = k.substring(SET_PREFIX.length)
-          if (newKey.startsWith(SYSTEM_PREFIX)) {
-            setModifiableConfig(sparkSession, newKey.substring(SYSTEM_PREFIX.length), v)
-          } else if (newKey.startsWith(HIVECONF_PREFIX) && newKey.startsWith(SPARK_PREFIX)) {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:61]: looks like a dead code path here

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 3]
```diff
@@ -53,26 +53,7 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
...
-            setModifiableConfig(sparkSession, newKey.substring(HIVECONF_PREFIX.length), v)
-          } else if (newKey.startsWith(HIVECONF_PREFIX)) {
-            setModifiableConfig(sparkSession, newKey.substring(HIVECONF_PREFIX.length), v)
-          } else if (newKey.startsWith(HIVEVAR_PREFIX) && newKey.startsWith(SPARK_PREFIX)) {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:65]: ditto

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 4]
```diff
@@ -97,6 +97,43 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
...
+        s" configuration, please remove it and retry if you want to proceed")
+    } else if (_confIgnoreList.contains(normalizedKey)) {
+      warn(s"$normalizedKey is a ignored key according to the sever side configuration")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:126]: ditto

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala [hunk 5]
```diff
@@ -53,26 +53,7 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
...
-            setModifiableConfig(sparkSession, newKey.substring(HIVEVAR_PREFIX.length), v)
-          } else if (newKey.startsWith(HIVEVAR_PREFIX)) {
-            setModifiableConfig(sparkSession, newKey.substring(HIVEVAR_PREFIX.length), v)
-          } else if (newKey.startsWith(METACONF_PREFIX) && newKey.startsWith(SPARK_PREFIX)) {
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala:69]: ditto

**Pull request closed without merging**